src/Aspire.Dashboard/Components/Pages/TraceDetail.razor.cs

```diff
@@ -242,6 +249,11 @@ private async Task OnToggleCollapse(SpanWaterfallViewModel viewModel)
             _collapsedSpanIds.Add(viewModel.Span.SpanId);
         }
 
+        if (_trace is not null)
+        {
+            _spanWaterfallViewModels = GetSpanWaterfallViewModels(_trace);
+        }
```

Comment:
Why not call `UpdateDetailViewData()` here? That's called in other places when page state is updated.